src/atoms/Heading/Heading.web.js

```diff
@@ -1,5 +1,96 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { getColorKeys, getColor } from '../../_helpers/theme';
+import baseTheme from '../../tokens/theme';
+import automation from '../../_helpers/automation-attributes';
 
-const Heading = () => <div>WIP</div>;
+import Text from '../Text';
+
+const styles = {
+  fontSize({ theme, size }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.size.medium;
+      case 'large':
+        return theme.fonts.size.large;
+      case 'xlarge':
+        return theme.fonts.size.xxlarge;
+      case 'xxlarge':
+        return theme.fonts.size.xxxlarge;
+      case 'xxxlarge':
+        return theme.fonts.size.xxxxlarge;
+      default:
+        return theme.fonts.size.xxxxlarge;
+    }
+  },
+  lineHeight({ theme, size, _weight }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.lineHeight.medium;
+      case 'large':
+        if (theme.fonts.weight[_weight] === theme.fonts.weight.bold) {
+          return theme.fonts.lineHeight.large;
+        }
+        return theme.fonts.lineHeight.medium;
+      case 'xlarge':
+        return theme.fonts.lineHeight.xlarge;
+      case 'xxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      case 'xxxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      default:
+        return theme.fonts.lineHeight.xxlarge;
+    }
+  },
+};
+
+const StyledHeadingText = styled(Text)`
```

Comment:
So finally we are not using the text component directly?

Reply:
Sorry about this. Actually Didn't really get this before when you were asking to change this.
I thought that you were asking to change the name of the component here.